src/dune_rules/opam_create.ml

```diff
@@ -108,6 +108,38 @@ let default_build_command =
            ~with_sites:Dune_project.(is_extension_set project dune_site_extension))
 ;;
 
+let depends_field t =
+  let depends = Package.depends t in
+  let pkg_already_in_depends (pkg : Package_dependency.t) =
+    List.exists
+      ~f:(fun (dep : Package_dependency.t) -> Package_name.equal pkg.name dep.name)
+      depends
+  in
+  let doc = t |> Package.info |> Package_info.documentation in
+  let doc_depends = doc.packages in
+  let doc_depends =
+    List.filter_map
+      ~f:(fun (doc_dep : Package_dependency.t) ->
+        if pkg_already_in_depends doc_dep
+        then None
+        else (
+          let constraint_ =
+            let c = Option.to_list doc_dep.constraint_ in
+            let c =
+              Package_constraint.And
+                (c
+                 @ [ Package_constraint.Bvar Package_variable_name.post
```

Comment:
I am slightly worried about adding `post` depends (mostly because I remember them being a headache, not because I have a specific issue in mind). Are we sure that declaring the dependencies as "post" is the correct semantic here?